### DISCUSSION TITLE
Welcome to Movie-Ticket-Booking Discussions!

### DISCUSSION THREAD

**Discussion by hoc081098** [hoc081098/Movie-Ticket-Booking#45] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by gunasekharravilla** [comment]
after a successful ticket booking ticket email is not received  

while checkout place the default user email  and phone number and allow edited as the user preference

**Reply by hoc081098** [reply]
Sometimes, Google automatically turns off "Allow less secure apps",  email cannot signed in and send email to user

**Reply by gunasekharravilla** [reply]
Is there any alternatives for successful mailing of ticket

**Reply by gunasekharravilla** [reply]
`import { google } from "googleapis";

const oauth2Client = new google.auth.OAuth2();

const accessToken = oauth2Client.getAccessToken()
@Module({
  imports: [
    MailerModule.forRootAsync({
      imports: [ConfigModule],
      inject: [ConfigService],
      useFactory: (configService: ConfigService) => {
        return {
          transport: {
            service: 'gmail',
            auth: {
              type: 'OAuth2',
              user: configService.get(ConfigKey.EMAIL), 
              clientId: configService.get(ConfigKey.clientId),
              clientSecret: configService.get(ConfigKey.clientSecret),
              refreshToken: configService.get(ConfigKey.refreshToken),
              accessToken: accessToken,
            },
            secure: false,
            ignoreTLS: true,
            rejectUnauthorized: false,
          },
          defaults: {`

Add related configs in env file

[reference](https://medium.com/@nickroach_50526/sending-emails-with-node-js-using-smtp-gmail-and-oauth2-316fe9c790a1)

**Comment by gunasekharravilla** [comment]


**Reply by hoc081098** [reply]
Thanks for reporting

**Reply by hoc081098** [reply]
I will fix it soon 😃

**Reply by hoc081098** [reply]
Already fixed https://github.com/hoc081098/Movie-Ticket-Booking/commit/0b1bb31f461ed6e49c497f735e08eb1281906f2f

**Comment by gunasekharravilla** [comment]
combo details booked by the user is not showing anywhere in the user app to show order details and collect the combo

**Comment by gunasekharravilla** [comment]
After successful  completion of payment, it's nice to redirect to a ticket confirmation page or view the ticket page

currently, it's redirecting to the showtimes page which will make the user select seats again for booking

**Comment by gunasekharravilla** [comment]
In some theaters, They have seat categories like Diamond, Gold, Silver  and its respective prices for it  

if you implement the seat price for every seat or  categorize seat layout like this look like a standard product

![image](https://user-images.githubusercontent.com/53616269/106564073-ba2a7380-6552-11eb-84d7-8be9d20ea278.png)

**Reply by gunasekharravilla** [reply]
if suppose seat in x==0  row alphabet has to appear on next  y axis

**Reply by hoc081098** [reply]
yeah, this can be implemented by adding a field `type: 'DIAMON' | 'GOLD' | 'SILVER'` to `seat.schema`

**Comment by gunasekharravilla** [comment]
Need help to generate a JW token  please guide me step by step to generate a token for this project @hoc081098 @phong016688 

anyone of you please guide me I have uncommented the code related token generation but I don't know which commands to use

**Reply by gunasekharravilla** [reply]
solved

**Comment by gunasekharravilla** [comment]
[![Codacy Badge](https://api.codacy.com/project/badge/Grade/4044596f649742fdb9b9c0acd80c321e)](https://www.codacy.com/app/zeshuaro/telegram-pdf-bot?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=zeshuaro/telegram-pdf-bot&amp;utm_campaign=Badge_Grade)

`https://www.codacy.com` add badge in the readme  and delete this comment after adding

**Reply by gunasekharravilla** [reply]
Do All the security check of the code in it

**Reply by hoc081098** [reply]
Thanks 🎉

**Comment by gunasekharravilla** [comment]
is it possible to create a web app from the same source code I have generated main.dart.js file  with 3lakhs LOC using `flutter web build` after that I have no idea how to move forward 

when I start `web serve`  shows a white blank screen

**Reply by hoc081098** [reply]
Some packages does not support WEB platform 😃

**Reply by gunasekharravilla** [reply]
I tried changing few packages to `flutter_web`  but many tutorials said there is a possibility to migrate the app to the web app

**Comment by gunasekharravilla** [comment]
Hey Petrus Could you help me how to setup neo4j in this App  please Share any reference link to set up it 

it shows an error `error: Slow Internet Occurred` and searching  is  not working 

could you explain how movieDB API is working  still now I didn't use that feature 😂😂

**Reply by hoc081098** [reply]
1. Download and install Neo4j Desktop https://neo4j.com/download/. Ref link: https://neo4j.com/developer/neo4j-desktop/

**Reply by hoc081098** [reply]
2. Launch Neo4j Desktop and Update your .env file: 
```
NEO4J_URL=...
NEO4J_USER=...
NEO4J_PASSWORD=...
```
You can take the url, user and password  from Neo4j desktop app

**Reply by hoc081098** [reply]
3. Transfer data from MongoDB to Neo4j: POST http://localhost:3000/neo4j/transfer.

**Reply by hoc081098** [reply]
recommendations using Neo4j: http://guides.neo4j.com/sandbox/recommendations

**Reply by hoc081098** [reply]
I uses MovieDb to insert seed data into database 😃

**Reply by gunasekharravilla** [reply]
In general, how neo4j URL looks like  for me I kept like this bolt+s://6e732647c204c0be454102ce2af3.neo4jsandbox.com:7474

**Reply by gunasekharravilla** [reply]
error i'm getting

`Neo4jError: Failed to connect to server. Please ensure that your database is listening on the correct host and port and that you have compatible encryption settings both on Neo4j server and driver. Note that the default encryption setting has changed in Neo4j 4.0. Caused by: Failed to establish connection in 30000ms`

**Reply by hoc081098** [reply]
my url is like this: `bolt://18.204.14.182:35023`

**Comment by gunasekharravilla** [comment]
Some theaters won't sell the combo based on the theater the combos and its prices will also differ  

good to implement the dynamic combos based on the theater and combos Not available widget also if the theater is not selling the combos 

for Extra User Interface we can keep veg or non-veg  symbols 😂

This is  big work I think for my knowledge for you it may be simple😂😂 

later we can develop for admin app to add products and edit products based on the client requirement and availability of the product

**Comment by gunasekharravilla** [comment]
[Source code](https://github.com/abuanwar072/Movie-Info---Flutter-UI)

![image](https://github.com/abuanwar072/Movie-Info---Flutter-UI/raw/master/movie_gif.gif)

Shall we keep this Carousel Animation for  Movies On theaters row? 

I tried to implement it, but all became messy 

Movie card Routing Also If a person taps on movie card, first open to movie detail page and after that After clicking A Button with `Book Tickets`  then route to the reservation page based on the movie or based on the theater

**Reply by gunasekharravilla** [reply]
```                
return AnimatedOpacity(
                  duration: Duration(milliseconds: 350),
                  opacity: initialPage == index ? 1 : 0.4,
                  child: Transform.rotate(
                    angle: math.pi * (1 * 0.038).clamp(-1, 1),
                    child: buildItem(
                      imageWidth,
                      imageHeight,
                      item,
                      context,
                      titleTextStyle,
                      reviewstextStyle,
                      minStyle,
                    ),
                  ),
```

what I did today which just rotated the card😏😏, I didn't understand how to modify page controller and initial state

**Reply by hoc081098** [reply]
I think we can convert the Horizontal list to a Carousel

**Reply by gunasekharravilla** [reply]
𝕐𝔼𝕊

**Reply by gunasekharravilla** [reply]
User prefer movie details when user clicks on movie card because after they want to know about movie cast and storyline ,ratings only some people will preferred to book ticket

**Comment by gunasekharravilla** [comment]
after now playing horizontal list or bottom of the home screen page shall we keep carousel like this for google ads

<details>

![Screenshot_2021-03-27-16-21-55-08](https://user-images.githubusercontent.com/53616269/112718523-9283d600-8f19-11eb-88c4-06025830a809.jpg)


</details>

**Comment by gunasekharravilla** [comment]
Hi hoc please check your LinkedIn
